fix logic in cdeps build

jedwards4b opened the pull request:
Fix logic in the cdeps build conditional 

Test suite: SMS_D_Ld5_Vnuopc.f10_f10_musgs.I2000Clm50BgcCropGs.cheyenne_intel.clm-default, SMS.f19_g17.X.cheyenne_intel
Test baseline: 
Test namelist changes: 
Test status: bit for bit,

Fixes #3684 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

mvertens commented:
Could you please verify that this works for the same compset with an MCT test as well.

mvertens commented:
I see you did this for the X compset - but it would be good to double check with the same I compset.

billsacks commented:
I don't feel able to give this a good review because I am not sure what the intended logic is here. So I'd say:
- If you are confident in the logic, then I'm fine approving this once the testing @mvertens has requested passes
- If you'd like a review of the logic, then can you describe what this is trying to accomplish?

jedwards4b commented:
The logic: if this is a nuopc build and cmeps_driver is None or this is a nuopc build and cmeps_driver is not 'nems'
then build cdeps.    Basically we need to build the cdeps library for all nuopc cases except when nems is the driver.

jedwards4b commented:
Test SMS_D_Ld5.f10_f10_musgs.I2000Clm50BgcCropGs.cheyenne_intel.clm-default also builds.

mvertens commented:
@jedwards4b - thank you for running this test.

On Mon, Aug 24, 2020 at 12:54 PM jedwards4b <notifications@github.com>
wrote:

> Test
> SMS_D_Ld5.f10_f10_musgs.I2000Clm50BgcCropGs.cheyenne_intel.clm-default also
> builds.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/3685#issuecomment-679304412>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB4XCE3VOWAAU3WJ2HMBXA3SCKZPLANCNFSM4QJZDJEA>
> .
>


-- 
Mariana Vertenstein
CESM Software Engineering Group Head
National Center for Atmospheric Research
Boulder, Colorado
Office 303-497-1349
Email: mvertens@ucar.edu

billsacks commented on scripts/lib/CIME/build.py:
```diff
@@ -355,7 +355,7 @@ def _build_libraries(case, exeroot, sharedpath, caseroot, cimeroot, libroot, lid
     # Build shared code of CDEPS nuopc data models
     cdeps_build_script = None
-    if comp_interface == "nuopc" and not cmeps_driver or not 'nems' in cmeps_driver:
+    if comp_interface == "nuopc" and (not cmeps_driver or not 'nems' in cmeps_driver):
```
Can you explain why you use `not 'nems' in cmeps_driver` instead of `cmeps_driver != 'nems'`?

jedwards4b commented:
@billsacks Because I didn't write the code.  Corrected and changed cmeps_driver to ufs_driver to be in line with the env variable.  Retested SMS_D_Ld5_Vnuopc.f10_f10_musgs.I2000Clm50BgcCropGs.cheyenne_intel.clm-default SMS_D_Ld5.f10_f10_musgs.I2000Clm50BgcCropGs.cheyenne_intel.clm-default
